src/row.rs

```diff
@@ -123,43 +142,127 @@ impl SnowflakeDecode for i8 {
 }
 
 impl SnowflakeDecode for String {
-    fn try_decode(value: &Option<String>) -> Result<Self> {
+    fn try_decode(value: &Option<String>, _: &SnowflakeColumnType) -> Result<Self> {
         let value = unwrap(value)?;
         Ok(value.to_string())
     }
 }
 
 impl SnowflakeDecode for bool {
-    fn try_decode(value: &Option<String>) -> Result<Self> {
+    fn try_decode(value: &Option<String>, _: &SnowflakeColumnType) -> Result<Self> {
         let value = unwrap(value)?;
-        if let Ok(v) = value.parse::<u16>() {
-            return Ok(v > 0);
-        }
-        if let Ok(v) = value.parse::<bool>() {
-            return Ok(v);
+        match value.to_uppercase().as_str() {
+            "1" | "TRUE" => Ok(true),
+            "0" | "FALSE" => Ok(false),
+            _ => Err(Error::Decode(format!("'{value}' is not bool"))),
         }
-        Err(Error::Decode(format!("'{value}' is not bool")))
     }
 }
 
 impl SnowflakeDecode for NaiveDateTime {
-    fn try_decode(value: &Option<String>) -> Result<Self> {
+    fn try_decode(value: &Option<String>, ty: &SnowflakeColumnType) -> Result<Self> {
         let value = unwrap(value)?;
-        if let Ok(v) = value.parse::<f64>() {
-            let secs = v.trunc() as i64;
-            let nsec = (v.fract() * 1_000_000_000.0) as u32;
-            let dt = DateTime::from_timestamp(secs, nsec)
-                .ok_or_else(|| Error::Decode(format!("invalid datetime: {}", value)))?;
-            return Ok(dt.naive_utc());
+        let scale = ty.scale.unwrap_or(9);
+        match ty.snowflake_type().to_ascii_uppercase().as_str() {
+            "TIMESTAMP_LTZ" | "TIMESTAMP_NTZ" => parse_timestamp_ntz_ltz(value, scale),
+            "TIMESTAMP_TZ" => parse_timestamp_tz(value, scale),
+            _ => Err(Error::Decode(format!(
+                "'{value}' is of type {}",
```

Comment:
The error message could be more user-friendly. It might be better to inform the user that the fetched timestamp type is not supported.

Reply:
fixed